tests/rules/test_quoted_strings.py

```diff
@@ -595,3 +595,338 @@ def test_allow_quoted_quotes(self):
                    "foo1: '[barbaz]'\n"
                    "foo2: '[bar\"baz]'\n",
                    conf)
+
+
+class QuotedKeysTestCase(RuleTestCase):
+    rule_id = 'quoted-strings'
+
+    def conf(self, options):
+        conf_with_options = (f"{self.rule_id}:\n"
+                             "  check-keys: true\n")
+        for option in options:
+            conf_with_options += f"  {option}\n"
+        return conf_with_options
+
+    key_strings = ('---\n'
+                   'true: 2\n'
+                   '123: 3\n'
+                   'foo1: 4\n'
+                   '"foo2": 5\n'
+                   '"false": 6\n'
+                   '"234": 7\n'
+                   '\'bar\': 8\n'
+                   '!!str generic_string: 9\n'
+                   '!!str 456: 10\n'
+                   '!!str "quoted_generic_string": 11\n'
+                   '!!binary binstring: 12\n'
+                   '!!int int_string: 13\n'
+                   '!!bool bool_string: 14\n'
+                   '!!bool "quoted_bool_string": 15\n'
+                   # Sequences and mappings
+                   '? - 16\n'
+                   '  - 17\n'
+                   ': 18\n'
+                   '[119, 219]: 19\n'
+                   '? a: 20\n'
+                   '  "b": 21\n'
+                   ': 22\n'
+                   '{a: 123, "b": 223}: 23\n'
+                   # Multiline strings
+                   '? |\n'
+                   '  line 1\n'
+                   '  line 2\n'
+                   ': 27\n'
+                   '? >\n'
+                   '  line 1\n'
+                   '  line 2\n'
+                   ': 31\n'
+                   '?\n'
+                   '  line 1\n'
+                   '  line 2\n'
+                   ': 35\n'
+                   '?\n'
+                   '  "line 1\\\n'
+                   '   line 2"\n'
+                   ': 39\n')
```

Comment:
I understand you propose to reuse a variable for testing various rule configurations. It is smart, but doing so has already been discussed in other pull requests and was finally rejected, because it makes it hard to read the tests and understand where it should fail / make sure there is no human error. (For instance, it's hard for me to review and visually understand/check test cases.)

I know it's a lot of code duplication, but inside tests we allow it if it helps for readability and maintenance.

Reply:
No problem. Readability and maintenance are important.